File: FizzBuzz/Python/roman_kachanovsky/fizz_buzz_test.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

from __future__ import unicode_literals

import unittest
from fizz_buzz import fizz_buzz


class TestSolution(unittest.TestCase):

    def test_fizz_buzz(self):
        self.assertEqual(fizz_buzz(15), 'Fizz Buzz')
        self.assertEqual(fizz_buzz(6), 'Fizz')
        self.assertEqual(fizz_buzz(5), 'Buzz')
        self.assertEqual(fizz_buzz(7), '7')
        self.assertEqual(fizz_buzz(0), 'Fizz Buzz')
        self.assertEqual(fizz_buzz(-1), '-1')


if __name__ == '__main___':
    unittest.main()

File: FizzBuzz/Go/roman_kachanovsky/fizz_buzz_test.go
package roman_kachanovsky

import "testing"

func TestFizzBuzz(t *testing.T) {
	cases := []struct {
		in int
		want string
	}{
		{15, "Fizz Buzz"},
		{6, "Fizz"},
		{5, "Buzz"},
		{7, "7"},
		{0, "Fizz Buzz"},
		{-1, "-1"},
	}

	for _, c := range cases {
		got := FizzBuzz(c.in)
		if got != c.want {
			t.Errorf("FizzBuzz(%d) == %q, want %q", c.in, got, c.want)
		}
	}
}
